source/fundamentals/crud/write-operations/change-a-document.txt

```diff
@@ -0,0 +1,203 @@
+
+   bson.D{{"<update operator>", bson.D{{"<field>", "<value>"}, 
+                                       {"<field>", "<value>"}, ... }}, 
+          {"<update operator>", ... }, ... }
+
+See the MongoDB server manual for a :manual:`complete list of update operators
+and descriptions </reference/operator/update-field/>`.
+
+.. note:: Aggregation Pipelines in Update Operations
+
+   If you are using MongoDB Version 4.2 or later, you can use aggregation
+   pipelines made up of a subset of aggregation stages in update operations. For
+   more information on the aggregation stages MongoDB supports in
+   aggregation pipelines, see our tutorial on performing
+   :manual:`updates with aggregation pipelines </tutorial/update-documents-with-aggregation-pipeline/>`.
+
+Example
+~~~~~~~
+
+The following document describes an employee:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Analyst",
+      "bonus" : 2500,
+      ...
+   }
+
+The following code shows how you can use ``UpdateOne()`` to:
+
+- Locate the document with ``_id`` value 2158.
+- Use the ``$set`` update operator to change the value of the ``name``
+  field to "Mary Wollstonecraft Shelley" and the ``role`` field to
+  "Marketing Director".
+- Use the ``$inc`` update operator to increment the value of the
+  ``bonus`` field by 2000.
+
+.. code-block:: go
+
+   filter := bson.D{{"_id", 2158}}
+   update := bson.D{{"$set", bson.D{{"name", "Mary Wollstonecraft Shelley"}, 
+       {"role", "Marketing Director"}}, {"$inc", bson.D{{"bonus", 2000}}}}
+
+   result, err := collection.UpdateOne(context.TODO(), filter, update)
+
+``UpdateOne()`` and ``UpdateMany`` return an ``UpdateResult`` type that
+contains information about the update operation such as
+``MatchedCount``, ``ModifiedCount``, ``UpsertedCount``, ``UpsertedID``.
+
+The following shows how to access the information in the
+``UpdateResult`` type:
+
+.. code-block:: go
+
+   fmt.Printf("Documents matched: %v\n", result.MatchedCount)
+   fmt.Printf("Documents modified: %v\n", result.ModifiedCount)
+   fmt.Printf("Documents upserted: %v\n", result.UpsertedCount)
+   fmt.Printf("_id of upserted document: %v\n", result.UpsertedID)
+
+The following shows the updated document resulting from the preceding update operation:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Wollstonecraft Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Director",
+      "bonus" : 4500,
+      ...
+   }
+
+If multiple documents match the query filter passed to ``UpdateOne()``,
+the method selects and updates the first matched document. If no
+documents match the query filter in the update operation,
+``UpdateOne()`` makes no changes to the document. See our **<TODO:
+link>** upsert guide to learn how to insert a new document if no
+documents match the query filter.
```

Comment:
Suggestion: I think this should be moved up too -- to before the example begins

Reply:
addressed